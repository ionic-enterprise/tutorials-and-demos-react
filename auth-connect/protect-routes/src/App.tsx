import { IonApp, IonRouterOutlet, setupIonicReact } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';
import { Redirect, Route } from 'react-router-dom';
import AuthActionCompletePage from './pages/AuthActionCompletePage';
import { AuthenticationProvider } from './providers/AuthenticationProvider';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/display.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/padding.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';

/* Theme variables */
import LoginPage from './pages/LoginPage';
import Tabs from './routes/Tabs';
import './theme/variables.css';
import { PrivateRoute } from './routes/PrivateRoute';

setupIonicReact();

const App: React.FC = () => (
  <IonApp>
    <AuthenticationProvider>
      <IonReactRouter>
        <IonRouterOutlet>
          <Route path="/tabs">
            <PrivateRoute>
              <Tabs />
            </PrivateRoute>
          </Route>
          <Route path="/auth-action-complete">
            <AuthActionCompletePage />
          </Route>
          <Route path="/login">
            <LoginPage />
          </Route>
          <Route exact path="/">
            <Redirect to="/tabs" />
          </Route>
        </IonRouterOutlet>
      </IonReactRouter>
    </AuthenticationProvider>
  </IonApp>
);

export default App;
