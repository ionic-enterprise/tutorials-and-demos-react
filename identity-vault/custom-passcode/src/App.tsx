import { IonApp, IonModal, IonRouterOutlet, setupIonicReact } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/display.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/padding.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';

import '@ionic/react/css/palettes/dark.system.css';

/* Theme variables */
import { useEffect, useRef, useState } from 'react';
import { Route } from 'react-router-dom';
import PinDialog from './components/PinDialog';
import Tabs from './components/Tabs';
import Login from './pages/Login';
import Start from './pages/Start';
import Unlock from './pages/Unlock';
import './theme/variables.css';
import { addOnPasscodeRequested, removeOnPasscodeRequested } from './util/session-vault';

setupIonicReact();

const App: React.FC = () => {
  const [showPinDialog, setShowPinDialog] = useState<boolean>(false);
  const passcodeSetRequest = useRef<boolean>(true);
  const passcodeCallback = useRef<((code: string) => void) | null>(null);

  useEffect(() => {
    const onPasscodeRequested = (isPasscodeSetRequest: boolean, onComplete: (code: string) => void) => {
      passcodeSetRequest.current = isPasscodeSetRequest;
      passcodeCallback.current = onComplete;
      setShowPinDialog(true);
    };
    addOnPasscodeRequested(onPasscodeRequested);
    return removeOnPasscodeRequested;
  }, []);

  return (
    <IonApp>
      <IonReactRouter>
        <IonModal isOpen={showPinDialog} backdropDismiss={false}>
          <PinDialog
            setPasscodeMode={passcodeSetRequest.current}
            onDismiss={(code: string | null) => {
              if (passcodeCallback.current) {
                passcodeCallback.current(code || '');
              }
              setShowPinDialog(false);
            }}
          />
        </IonModal>

        <IonRouterOutlet>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/">
            <Start />
          </Route>
          <Route exact path="/unlock">
            <Unlock />
          </Route>
          <Route path="/tabs">
            <Tabs />
          </Route>
        </IonRouterOutlet>
      </IonReactRouter>
    </IonApp>
  );
};

export default App;
