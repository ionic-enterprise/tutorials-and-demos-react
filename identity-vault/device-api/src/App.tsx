import { IonApp, IonRouterOutlet, setupIonicReact } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/display.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/padding.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';

import '@ionic/react/css/palettes/dark.system.css';

/* Theme variables */
import Tabs from './components/Tabs';
import './theme/variables.css';
import Login from './pages/Login';
import { Route } from 'react-router-dom';
import Start from './pages/Start';
import Unlock from './pages/Unlock';

setupIonicReact();

const App: React.FC = () => (
  <IonApp>
    <IonReactRouter>
      <IonRouterOutlet>
        <Route exact path="/login">
          <Login />
        </Route>
        <Route exact path="/">
          <Start />
        </Route>
        <Route exact path="/unlock">
          <Unlock />
        </Route>
        <Route path="/tabs">
          <Tabs />
        </Route>
      </IonRouterOutlet>
    </IonReactRouter>
  </IonApp>
);

export default App;
