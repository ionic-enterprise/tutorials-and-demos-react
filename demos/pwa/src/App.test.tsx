import { vi } from 'vitest';
import { render, waitFor } from '@testing-library/react';
import App from './App';

vi.mock('./utils/auth');

describe('<App />', () => {
  it('renders without crashing', async () => {
    const { baseElement } = render(<App />);
    await waitFor(() => expect(baseElement).toBeDefined());
  });

  it('renders consistently', async () => {
    const { asFragment } = render(<App />);
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });
});
