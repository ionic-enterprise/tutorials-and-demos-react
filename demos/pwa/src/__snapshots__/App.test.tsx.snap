// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > renders consistently 1`] = `
<DocumentFragment>
  <ion-app>
    <ion-spinner />
  </ion-app>
</DocumentFragment>
`;
